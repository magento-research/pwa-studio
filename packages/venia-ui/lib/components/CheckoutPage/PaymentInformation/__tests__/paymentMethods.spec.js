import React from 'react';
import createTestInstance from '@magento/peregrine/lib/util/createTestInstance';

import PaymentMethods from '../paymentMethods';
import CreditCardPaymentMethod from '../creditCardPaymentMethod';

jest.mock('../../../../classify');

jest.mock('../creditCardPaymentMethod', () => () => (
    <div>Credit Card Payment Method Component</div>
));

jest.mock(
    '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentMethods',
    () => ({
        usePaymentMethods: jest.fn().mockReturnValue({})
    })
);

test('Should return correct shape', () => {
    const tree = createTestInstance(
        <PaymentMethods selectedPaymentMethod="creditCard" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should render creditCardPaymentMethod component with isHidden prop set to false if selectedPaymentMethod is creditCard', () => {
    const tree = createTestInstance(
        <PaymentMethods selectedPaymentMethod="creditCard" />
    );

    expect(
        tree.root.findByType(CreditCardPaymentMethod).props.isHidden
    ).toBeFalsy();
});
