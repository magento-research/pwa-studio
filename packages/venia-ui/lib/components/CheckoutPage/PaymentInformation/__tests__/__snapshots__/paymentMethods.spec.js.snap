// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return correct shape 1`] = `
<div
  className="root"
>
  <div
    className="payment_method"
  >
    <label
      className="root"
    >
      <input
        checked={false}
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="creditCard"
      />
      <span
        className="label"
      >
        Credit Card
      </span>
    </label>
    <div>
      Credit Card Payment Method Component
    </div>
  </div>
</div>
`;
